### PR TITLE
workspaceId should be part of spec request

### LINKED ISSUE #12139: Spec API endpoint should contain WorkspaceID
For BigQuery Airbyte staging, we need to figure out the workspace that is requesting for the spec of a connector so that we can display the service account email associated with that workspace. Currently the spec endpoint doesnt accept the workspaceId. We need to change that and make workspaceID part of the API call

### PR DESCRIPTION
**Pull request by subodh1810**
Issue : https://github.com/airbytehq/airbyte/issues/12139

### CODE REVIEW

File: airbyte-webapp/src/services/connector/DestinationDefinitionSpecificationService.tsx [hunk 1]
```diff
@@ -28,17 +29,19 @@ function useGetService(): DestinationDefinitionSpecificationService {
...
 
-  return useSuspenseQuery(destinationDefinitionSpecificationKeys.detail(id), () => service.get(id));
+  return useSuspenseQuery(destinationDefinitionSpecificationKeys.detail(id), () => service.get(id, workspaceId));
 };
 
 export const useGetDestinationDefinitionSpecificationAsync = (
   id: string | null
 ): QueryObserverResult<DestinationDefinitionSpecification, Error> => {
   const service = useGetService();
+  const workspaceId = useCurrentWorkspace().workspaceId;
```
Comment (timroes) [inline airbyte-webapp/src/services/connector/DestinationDefinitionSpecificationService.tsx:41]:
```suggestion
  const { workspaceId } = useCurrentWorkspace();
```

File: airbyte-webapp/src/services/connector/SourceDefinitionSpecificationService.tsx [hunk 2]
```diff
@@ -28,17 +29,19 @@ function useGetService(): SourceDefinitionSpecificationService {
 
 export const useGetSourceDefinitionSpecification = (id: string): SourceDefinitionSpecification => {
   const service = useGetService();
+  const workspaceId = useCurrentWorkspace().workspaceId;
```
Comment (timroes) [inline airbyte-webapp/src/services/connector/SourceDefinitionSpecificationService.tsx:32]:
```suggestion
  const { workspaceId } = useCurrentWorkspace();
```

File: airbyte-webapp/src/services/connector/SourceDefinitionSpecificationService.tsx [hunk 3]
```diff
@@ -28,17 +29,19 @@ function useGetService(): SourceDefinitionSpecificationService {
...
 
-  return useSuspenseQuery(sourceDefinitionSpecificationKeys.detail(id), () => service.get(id));
+  return useSuspenseQuery(sourceDefinitionSpecificationKeys.detail(id), () => service.get(id, workspaceId));
 };
 
 export const useGetSourceDefinitionSpecificationAsync = (
   id: string | null
 ): QueryObserverResult<SourceDefinitionSpecification, Error> => {
   const service = useGetService();
+  const workspaceId = useCurrentWorkspace().workspaceId;
```
Comment (timroes) [inline airbyte-webapp/src/services/connector/SourceDefinitionSpecificationService.tsx:41]:
```suggestion
  const { workspaceId } = useCurrentWorkspace();
```

**Review by timroes — APPROVED** [review]
FE code looks good to me awaiting some minor code style improvements. Haven't tested locally.

**Review by tuliren — APPROVED** [review]

**Review by edgao — APPROVED** [review]

**Review by alafanechere — APPROVED** [review]
I had to refactor a bit the CLI because it's using the endpoint that was modified in this PR and some signatures in the generated API client changed.

**Pull request merged**